feature-libs/product-configurator/rulebased/cpq/cpq-configurator-normalizer.ts

```diff
@@ -0,0 +1,407 @@
+import { Injectable } from '@angular/core';
+import { Converter, TranslationService } from '@spartacus/core';
+import { take } from 'rxjs/operators';
+import { Configurator } from './../core/model/configurator.model';
+import { CpqConfiguratorUtilsService } from './cpq-configurator-utils.service';
+import { Cpq } from './cpq.models';
+
+@Injectable()
+export class CpqConfiguratorNormalizer
+  implements Converter<Cpq.Configuration, Configurator.Configuration> {
+  constructor(
+    protected cpqUtilitiesService: CpqConfiguratorUtilsService,
+    protected translation: TranslationService
+  ) {}
+
+  convert(
```

Comment:
Only `convert` needs to be public in converter class. The rest of methods can be `protected`.
You can access the protected properties in the tests without loosing type safety `object['methodName']()`.
See https://stackoverflow.com/questions/35987055/how-to-write-unit-testing-for-angular-typescript-for-private-methods-with-jasm

Reply:
Pick

Reply:
Done! See #11842